packages/devtools-sham-modules/client/framework/menu.js

```diff
@@ -0,0 +1,176 @@
+/* -*- indent-tabs-mode: nil; js-indent-level: 2 -*- */
```

Comment:
this is almost a direct copy from m-c. I took a lot of pains to move the *sham* work to `utils/menu`. A regular `menuitem` will behave like a context menu in a web context.

Three things changed:
1. i referenced the sham events emitter (we should consider de-priviledging events emitter a top priority, it's really close)
2. I added `popup` as a argument to the close and open events
3. I created a new `createPopup` function that's overridden in the web context.